### ISSUE TITLE
multi_scan is not passing along the :only_testing option

### ISSUE THREAD

**Issue by lyndsey-ferguson** [lyndsey-ferguson/fastlane-plugin-test_center#32]
When `multi_scan` is given the `:only_testing` option with a test target, it is not passing that along to the `scan` tool and the result is that all tests in all test targets is tested instead of just the tests in one test target.

See Issue #31

**Comment by lyndsey-ferguson** [comment]
I have confirmed this with my sample `AtomicBoy` project.

**Comment by lyndsey-ferguson** [comment]
I am stripping out that option, not sure why yet.

**Comment by lyndsey-ferguson** [comment]
`TestCollector` is looking for test identifiers that match the pattern of `test-target/testsuite` and we are giving it only `test-target`. I believe that `xcodebuild` can match against just `test-target`.

**Comment by lyndsey-ferguson** [comment]
@NXGieng, FYI, I am closing in a fix too. The best way of fixing this issue, in my opinion, is in the `TestCollector's` `only_testing_to_testables_tests` method. It should be able to handle a `test_identifier` of one, two, or three components.

Currently, it is expecting more than one.

**Comment by lyndsey-ferguson** [comment]
After a few false starts in my investigation, it seems that `TestCollector` is providing the correct test identifiers, however, how I handle the results in `correcting_scan_helper`, `scan_testable` method is not catching the case where I have multiple possible testables, but want only one.

**Comment by lyndsey-ferguson** [comment]
It would seem to me, that I want an empty list for the other testables. So, `test_collector` should look at the `xctestrun` file and gather the other testables and create an empty list for those.

**Comment by lyndsey-ferguson** [comment]
@NXGieng look at [this branch](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/tree/fix-issue-32-scan-not-receiving-only-testing) to see how I think we should start the approach to fix this. There is a problem where I am trying to slice an empty array which I think we can check for.

**Comment by NXGieng** [comment]
Hi @lyndsey-ferguson , Nice work! 
There are two things that I want to share for your commit. About the function, I commented it bellow, 
```ruby
def only_testing_to_testables_tests
  tests = Hash.new { |h, k| h[k] = [] }
  @only_testing.sort.each do |test_identifier|   # can we remove sort here? as we might need to prioritise some test before others. 
    testable = test_identifier.split('/', 2)[0]
    tests[testable] << test_identifier
  end
  config = FastlaneCore::Configuration.create(::Fastlane::Actions::TestsFromXctestrunAction.available_options, xctestrun: @xctestrun_path)
  all_tests = ::Fastlane::Actions::TestsFromXctestrunAction.run(config)
  all_tests.each_key { |testable| tests[testable] = [] unless tests.has_key?(testable) } 
  # ^ So we are cosidering only_testing is now used for test targets, if we use it for test suites, will it be incorrect? let's say we want to test for "AtomicBoyTests/Test1" only, but this will include "AtomicBoyTests/Test2" as well because they have the same key "AtomicBoyTests".
  tests
end
```

**Comment by lyndsey-ferguson** [comment]
We could remove the sort, at first, I wanted all tests for one testable to be done before the next one, but that really doesn't matter.

With respect to your second question, that you would get a `Hash` of:
```
{
  'AtomicBoyTests' => [ 'AtomicBoyTests/test2' ],
  'AtomicBoyUITests' => []
}
```
If `multi_scan` is provided the `:only_testing` option of `[ 'AtomicBoyTests/test2' ]` as the code `@only_testing.sort.each do |test_identifier|` would only see that one test, and the code, `all_tests.each_key { ... }` would _only add_ new testables if it doesn't already exist (which in my example is `AtomicBoyUITests`.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.0.6](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/tree/v3.0.6)
